src/components/BaseCard.jsx

```diff
@@ -0,0 +1,39 @@
+import { Thumbnail } from "./Thumbnail";
+import { useNavigate } from "react-router-dom";
+
+export function BaseCard({ children, id, avatarSrc, ...rest }) {
+  const navigate = useNavigate();
+  const openVideo = (e) => {
+    navigate(`/video/${id}`);
+    e.stopPropagation();
+  };
+
+  return (
+    <div key={id} className="card card--shadow m-1">
+      <Thumbnail onClick={openVideo} id={id} className="clickable" />
```

Comment:
The `onClick={openVideo}` function should be on the upper div and not on Thumbnail, due to which clicking on an avatar or video title, the video is not opening up.